SimpleExec/ProcessExtensions.cs

```diff
@@ -1,16 +1,54 @@
 namespace SimpleExec
 {
     using System;
+    using System.Text;
     using System.Diagnostics;
     using System.Threading;
     using System.Threading.Tasks;
 
     internal static class ProcessExtensions
     {
-        public static void Run(this Process process, bool noEcho, string echoPrefix)
+        public static string Run(this Process process, bool noEcho, string echoPrefix, bool readStandardOutput, CancellationToken cancellationToken)
         {
-            process.EchoAndStart(noEcho, echoPrefix);
-            process.WaitForExit();
+            using (var resetEvent = new ManualResetEventSlim(false))
+            {
+                StringBuilder stringBuilder = null;
+                process.Exited += (s, e) => resetEvent.SafeSet();
+                if(readStandardOutput)
+                {
+                    stringBuilder = new StringBuilder();
+                    process.OutputDataReceived += (s, e) => stringBuilder.AppendLine(e.Data);

```

Comment:
Once we have settled on how you want this code to look like we could apply the same to the async path if desired for consistency